### PR TITLE
Make default testcase filter property name FullyQualifiedName 

### PR DESCRIPTION
**Pull request by smadala**
- Convert `dotnet test --filter "ABC"` to `dotnet test --filter "FullyQualifiedName~ABC"`.
Related to: https://github.com/Microsoft/vstest/issues/334

### CODE REVIEW

File: test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs [hunk 1]
```diff
@@ -128,5 +128,26 @@ public void RunSelectedTestsWithPriorityTrait(string runnerFramework, string tar
...
         }
+
+        /// <summary>
+        /// In case TestCaseFilter is provide without any property like Name or ClassName. ex. /TestCaseFilter:"UnitTest1"
+        /// this command should provide same results as /TestCaseFilter:"FullyQualifiedName~UnitTest1".
+        /// </summary>
+        [CustomDataTestMethod]
+        [NET46TargetFramework]
+        [NETCORETargetFramework]
+        public void TestCaseFilterShouldWork_IfOnlyPropertyValueGivenInExpression(string runnerFramework, string targetFramework, string targetRuntime)
```
Comment (codito) [inline test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs]: nit: please remove underscore

File: src/Microsoft.TestPlatform.Common/Filtering/Condition.cs [hunk 2]
```diff
@@ -156,17 +165,28 @@ internal bool Evaluate(Func<string, Object> propertyValueProvider)
         /// </summary>
         internal static Condition Parse(string conditionString)
         {
+            if (string.IsNullOrWhiteSpace(conditionString))
+            {
+                ThrownFormatExceptionForInvalidCondition(conditionString);
+            }
             string[] parts = Regex.Split(conditionString, propertyNameValueSeperatorString);
+            if (parts.Length == 1)
```
Comment (codito) [inline src/Microsoft.TestPlatform.Common/Filtering/Condition.cs]: Isn't there an unit test for this?

**Review by codito — APPROVED** [review]
LGTM with few suggestions.

File: src/Microsoft.TestPlatform.Common/Filtering/Condition.cs (hunk 2 above)
Comment (smadala) [inline src/Microsoft.TestPlatform.Common/Filtering/Condition.cs]: test is added in `ConditionTests.cs`.

File: test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs [hunk 3]
```diff
@@ -128,5 +128,26 @@ public void RunSelectedTestsWithPriorityTrait(string runnerFramework, string tar
...
+        {
+            AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerFramework, targetFramework, targetRuntime);
```
Comment (AbhitejJohn) [inline test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs]: nit: Looks repetitive. Move to constructor/Test Init?

File: test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            var conditionString = "";
+            Assert.ThrowsException<FormatException>(() => Condition.Parse(conditionString));
+        }
+
+        [TestMethod]
+        public void ParseShouldThrownFormatException_OnIncompleteConditionString()
+        {
+            var conditionString = "PropertyName=";
+            Assert.ThrowsException<FormatException>( ()=> Condition.Parse(conditionString));
+        }
```
Comment (AbhitejJohn) [inline test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs]: nit: space.

File: test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+        [TestMethod]
+        public void ParseShouldCreateDefaultCondition_WhenOnlyPropertyValuePassed()
```
Comment (AbhitejJohn) [inline test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs]: nit: no underscores.

**Review by AbhitejJohn — APPROVED** [review]

**Comment by codito** [comment]
Related https://github.com/Microsoft/vstest/issues/334

File: test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs (hunk 3 above)
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs]: We are using data row here, can't call from .ctor/Setup.

File: test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs (hunk 4 above)
Comment (smadala) [inline test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs]: Done 👍

File: test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs (hunk 5 above)
Comment (smadala) [inline test/Microsoft.TestPlatform.Common.UnitTests/Filtering/ConditionTests.cs]: Done 👍

File: test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs (hunk 1 above)
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs]: Done 👍

**Pull request merged**